roles/build/tasks/build_remote.yml

```diff
@@ -0,0 +1,195 @@
+    remote_build_host: "{{ (groups['cn'] + groups['pn'] + groups['en'])[0] }}"
+
+- name: Build Remote - Install build dependencies
+  block:
+    - name: Update package cache
+      ansible.builtin.shell: |
+        sudo yum makecache
+      args:
+        executable: /bin/bash
+
+    - name: Install essential build tools
+      ansible.builtin.shell: |
+        sudo yum install -y git make gcc curl wget unzip which
+      args:
+        executable: /bin/bash
+
+    - name: Check if Go is installed
+      ansible.builtin.command: which go
+      register: go_check
+      ignore_errors: yes
+      changed_when: false
+
+    - name: Install Go if not present
+      block:
+        - name: Download Go
+          ansible.builtin.get_url:
+            url: "https://golang.org/dl/go1.23.0.linux-amd64.tar.gz"
+            dest: "~/go1.23.0.linux-amd64.tar.gz"
+            mode: '0644'
+          become: yes
+
+        - name: Extract Go
+          ansible.builtin.unarchive:
+            src: "~/go1.23.0.linux-amd64.tar.gz"
+            dest: "/usr/local"
+            remote_src: yes
+          become: yes
+
+        - name: Add Go to PATH
+          ansible.builtin.lineinfile:
+            path: "~/.bashrc"
+            line: 'export PATH=$PATH:/usr/local/go/bin'
+            state: present
+
+        - name: Set Go PATH for current session
+          ansible.builtin.set_fact:
+            ansible_env:
+              PATH: "{{ ansible_env.PATH }}:/usr/local/go/bin"
+      when: go_check.rc != 0
+  delegate_to: "{{ remote_build_host }}"
+  when: kaia_build_required
+
+- name: Build Remote - Create remote build directory
+  ansible.builtin.file:
+    path: "~/kaia_build"
+    state: directory
+    mode: '0755'
+  delegate_to: "{{ remote_build_host }}"
+  when:
+    - kaia_build_required
+
+- name: Build Remote - Clean remote build directory
+  ansible.builtin.file:
+    state: absent
+    path: "~/kaia_build"
+  delegate_to: "{{ remote_build_host }}"
+  when:
+    - kaia_build_required
+
+- name: Build Remote - Git checkout
+  ansible.builtin.git:
+    repo: "{{ kaia_build_remote_git_url }}"
+    dest: "~/kaia_build"
+    version: "{{ kaia_build_remote_git_branch }}"
+  delegate_to: "{{ remote_build_host }}"
+  when:
+    - kaia_build_required
+
+- name: Build Remote - Compile kaia binary
+  ansible.builtin.command: make all
+  args:
+    chdir: "~/kaia_build"
+  delegate_to: "{{ remote_build_host }}"
+  when:
+    - kaia_build_required
+
+- name: Build Remote - Git checkout for homi (local)
+  ansible.builtin.git:
+    repo: "{{ kaia_build_remote_git_url }}"
+    dest: "{{ kaia_build_dir }}"
+    version: "{{ kaia_build_remote_git_branch }}"
+  when:
+    - kaia_build_required
+
+- name: Build Remote - Compile homi binary locally
+  ansible.builtin.command: |
+    make homi
+  args:
+    chdir: "{{ kaia_build_dir }}"
+  when:
+    - kaia_build_required
+
+- name: Build Remote - Copy homi binary
+  ansible.builtin.copy:
+    dest: "{{ bin_dir }}/homi"
```

Comment:
Why needed compile on local for homi and copy from it?

Is it enough to do this? Would you copy from remote to local?
* https://github.com/kaiachain/kaiaspray/pull/63/files#diff-0356d6f2f1258f0256f8538d463949a60dc75c97bee6efe75bc39340089a0356R95
```
- name: Build Remote - Compile kaia binary
  ansible.builtin.command: make all
  args:
    chdir: "~/kaia_build"
  delegate_to: "{{ remote_build_host }}"
  when:
    - kaia_build_required
```